homeassistant/components/citybikes/sensor.py

```diff
@@ -116,7 +118,12 @@
         vol.Required(ATTR_NAME): cv.string,
         vol.Required(ATTR_TIMESTAMP): cv.string,
         vol.Optional(ATTR_EXTRA): vol.Schema(
-            {vol.Optional(ATTR_UID): cv.string}, extra=vol.REMOVE_EXTRA
+            {
+                vol.Optional(ATTR_UID): cv.string,
+                vol.Optional(ATTR_HAS_EBIKES): cv.boolean,
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
Hello! Thank you for reviewing!

I am a little confused, I don't understand why adding support for e-bike data has to be blocked by this PyPi requirement. To me, it looks like this component was already breaking that PyPi requirement completely. So I would think there should be an issue open on the repo to refactor the whole CityBikes component, but I couldn't find any such issue.

So it feels like you are saying HA should block features and improvements because of, what seems like to me, out-of-scope tech debt, which isn't even documented as an open issue in the repo. Please let me know if I'm wrong.

Reply:
No, you are right; we block it because of tech debt, and prevent more debt to be introduced.